backend/Tasque.WebAPI/Controllers/EntityController.cs

```diff
@@ -0,0 +1,78 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Tasque.Core.BLL.Services;
+using Tasque.Core.Common.Entities.Abstract;
+
+namespace Tasque.Core.WebAPI.Controllers
+{
+    [ApiController]
+    public abstract class EntityController<TModel, TDto, TService> : ControllerBase
+        where TModel : BaseEntity
+        where TDto : class
+        where TService : IEntityCrudService<TModel>
+    {
+        private readonly TService _service;
+        private readonly Mapper mapper;
+        public EntityController(TService service)
+        {
+            _service = service;
+            var config = new MapperConfiguration(cfg => cfg.CreateMap<TDto, TModel>().ReverseMap());
```

Comment:
we may have specific mapper profiles so I think it's better to pull IValidator<TModel> from DI container

Reply:
same goes for validators

Reply:
Will be investigated and implemented as a standalone task